### DISCUSSION TITLE
NextJS values from config are not working.

### DISCUSSION THREAD

**Discussion by dawidseipold** [master-co/css#76] [Help] [answered]
Hi, so I just discovered Master CSS and I really like it. However, I just can't seem to make it work with NextJS properly. It works out of the box, but when I try to initialize the custom config, it just ignores it. Here's my code:

src/pages/_document.ts
```ts
import Document, { Html, Head, Main, NextScript, DocumentContext } from 'next/document';
import { render } from '@master/css/render';
import { StyleSheet } from '@master/css';

export const MyDocument = () => {
  return (
    <Html>
      <Head>
        <script dangerouslySetInnerHTML={{ __html: 'window.MasterCSSManual=true' }} />
      </Head>
      <body>
        <Main />
        <NextScript />
      </body>
    </Html>
  );
};

MyDocument.getInitialProps = async (ctx: DocumentContext) => {
  const { css } = render((await ctx.renderPage()).html, { StyleSheet });
  const initialProps = await Document.getInitialProps(ctx);

  return {
    ...initialProps,
    styles: (
      <>
        <style id="master-css">{css}</style>
        {initialProps.styles}
      </>
    ),
  };
};

export default MyDocument;
```

src/styles/master.css.ts
```ts
import { init, Style } from '@master/css';

Style.colorSchemes.push('hc');
Style.extend('classes', {
  btn: 'font:14 h:40 h:32.sm text:center bg:red:hover',
  card: 'p:20 b:1|solid|gray-80 bg:white',
});

init();
```

src/pages/index.ts
```ts
import type { NextPageWithLayout } from './_app';

const Home: NextPageWithLayout = () => {
  return (
    <div className="color:yellow-70@test color:red-60@hc color:beryl-60@light color:blue-60@dark btn">
      Hello World!
    </div>
  );
};

export default Home;
```

For example, in the index file, it just ignores the ``hc`` color scheme and default for the first defined one (``test`` in this case). Likewise, it completely ignores the ``btn`` class.

**Comment by 1aron** [comment, marked as answer]
@dawidseipold Did you have import `src/styles/master.css.ts` into `_pages/_app.tsx`?
```ts
 import '../styles/master.css'
 ```

**Reply by dawidseipold** [reply]
Right! I totally forgot about replacing the import from the default one! Thank you for quick response 🙂

**Reply by 1aron** [reply]
Anytime! I think I'll write each step out in 2.0. 🤔

**Reply by 1aron** [reply]
Add `hc` class name at `<html class=Name"hc">`?

**Reply by dawidseipold** [reply]
I did that. It was something else on my end, but it's fixed now. Hence, why I removed that comment. Anyway, thank you for help!

**Reply by 1aron** [reply]
The automatically generated CSS rules look like this:
<img width="632" alt="Screen Shot 2022-10-08 at 9 00 50 PM" src="https://user-images.githubusercontent.com/33840671/194708848-c6374883-e495-4f39-87ed-db9a651a532e.png">

**Reply by 1aron** [reply]
@dawidseipold If you want to get my reply asap. Join our discord channel: https://discord.com/invite/sZNKpAAAw6 😄
